master/internal/telemetry/reports.go

```diff
@@ -19,17 +18,9 @@ import (
 )
 
 // ReportMasterTick reports the master snapshot on a periodic tick.
-func ReportMasterTick(system *actor.System, db db.DB, rm telemetryRPFetcher) {
+func ReportMasterTick(resp *apiv1.GetResourcePoolsResponse, db db.DB) {
 	resourceManagerType := ""
 
-	req := &apiv1.GetResourcePoolsRequest{}
-	resp, err := rm.GetResourcePools(system, req)
-	if err != nil {
-		// TODO(Brad): Make this routine more accepting of failures.
-		logrus.WithError(err).Error("failed to receive resource pool telemetry information")
-		return
-	}
```

Comment:
imo it's cleaner to keep the `system` variable & use it in the same function, `tick`. I don't even have `system` as part of the Telemeter struct for this reason. It's only used once, in one function. Also, in the future it would be awesome to factor out `telemetryRPFetcher`,  which now only shows up in `telemetry.go` and not `reports.go` too.

Reply:
i just figure that in the next 3-6mo `rm` will be a global singleton and it won't take a `system`

Reply:
but that is hopeful, ha